packages/vaadin-select/src/vaadin-select.js

```diff
@@ -358,7 +358,7 @@ class SelectElement extends ElementMixin(
     this._nativeInput.setAttribute('tabindex', -1);
     this._nativeInput.style.pointerEvents = 'none';
 
-    this.focusElement.addEventListener('click', () => (this.opened = !this.readonly));
+    this.focusElement.addEventListener('click', (e) => (this.opened = !this.readonly && !e.defaultPrevented));
```

Comment:
This can be reverted, too.

Reply:
We can remove extra event listener and use the following code to detect helper click:

```js
    this.focusElement.addEventListener('click', (e) => {
      const isHelperClick = Array.from(e.composedPath()).some((node) => {
        return node.nodeType === Node.ELEMENT_NODE && node.getAttribute('slot') === 'helper';
      });
      this.opened = !this.readonly && isHelperClick;
    });
```

Reply:
I applied the suggested change it tests are working good, I don't know if there is enough test for this case or not though.

Reply:
Thanks, the test that you already added looks good to me and covers the new solution.